#!/usr/bin/env python
# -*- coding: utf-8 -*-
PACKAGE = "pose_recovery_pkg"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("map_frame_id", str_t, 0, "Map Frame",  "map")
gen.add("base_frame_id", str_t, 0, "Base Frame",  "base_link")

gen.add("file_name", str_t, 0, "roslaunch的保存路徑在~/.ros, rosrun的保存路徑在執行指令的終端機目錄",  "robot_pose");
gen.add("period", double_t, 0, "保存的週期 (s)", 0.5, 0.1, 2.0)
gen.add("file_buffer",int_t, 0, "為了避免寫檔沒有完成時就斷電, 必須要有2個以上的緩衝交替寫", 2,  2, 5)

gen.add("covariance_x", double_t, 0, "x方向的分佈 (m)", 0.25, 0.01, 2.0)
gen.add("covariance_y", double_t, 0, "y方向的分佈 (m)", 0.25, 0.01, 2.0)
gen.add("covariance_theta", double_t, 0, "角度的分佈 (deg)", 40.0, 0.0, 180.0)

gen.add("initial", bool_t, 0, "(測試)重新發出位置估測訊息", False)

gen.add("save", bool_t, 0, "save parameters", False)
gen.add("reset", bool_t, 0, "reset parameter", False)

exit(gen.generate(PACKAGE, "pose_recovery_pkg", "pose_recovery_parameters"))
